code/datums/gamemode/factions/bloodcult/bloodcult_mobs_and_constructs.dm

```diff
@@ -335,3 +335,394 @@
+var/list/astral_projections = list()
+
+/mob/living/simple_animal/astral_projection
+	name = "astral projection"
+	real_name = "astral projection"
+	desc = "A fragment of a cultist's soul, freed from the laws of physics."
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "ghost-narsie"
+	icon_living = "ghost-narsie"
+	icon_dead = "ghost-narsie"
+	maxHealth = 1
+	health = 1
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	minbodytemp = 0
+	maxbodytemp = 4000
+	min_oxy = 0
+	max_co2 = 0
+	max_tox = 0
+	speed = 1
+	stop_automated_movement = TRUE
+	faction = "cult"
+	supernatural = TRUE
+	flying = TRUE
+	mob_property_flags = MOB_SUPERNATURAL
+	speed = 0.5
+	forced_density = 1
+	density = 0
+	lockflags = 0
+	canmove = 0
+	blinded = 0
+	anchored = 1
+	flags = HEAR | TIMELESS | INVULNERABLE
+	universal_understand = 1
+	universal_speak = 1
+	plane = GHOST_PLANE
+	layer = GHOST_LAYER
+	invisibility = INVISIBILITY_CULTJAUNT
+	see_invisible = SEE_INVISIBLE_OBSERVER_NOLIGHTING
+	incorporeal_move = INCORPOREAL_GHOST
+	alpha = 127
+
+	var/tangibility = FALSE//keeps track of whether we're in "ghost" form or "slightly less ghost" form
+
+	var/mob/living/anchor//the cultist's original body
+	var/image/incorporeal_appearance
+	var/image/tangible_appearance
+
+	var/time_last_speech = 0//speech bubble cooldown
+
+	//sechud stuff
+	var/cardjob = "hudunknown"
+
+	//convertibility HUD
+	var/list/propension = list()
+
+
+/mob/living/simple_animal/astral_projection/New()
+	..()
+	astral_projections += src
+
+	incorporeal_appearance = image('icons/mob/mob.dmi',"blank")
+	tangible_appearance = image('icons/mob/mob.dmi',"blank")
+	change_sight(adding = SEE_TURFS | SEE_MOBS | SEE_OBJS | SEE_SELF)
+	see_in_dark = 100
+	hud_list[ID_HUD]          = image('icons/mob/hud.dmi', src, "hudunknown")
+	add_spell(new /spell/astral_return, "cult_spell_ready", /obj/abstract/screen/movable/spell_master/bloodcult)
+	add_spell(new /spell/astral_toggle, "cult_spell_ready", /obj/abstract/screen/movable/spell_master/bloodcult)
+
+/mob/living/simple_animal/astral_projection/Login()
+	..()
+	client.CAN_MOVE_DIAGONALLY = 1
+
+	if (!tangibility)
+		overlay_fullscreen("astralborder", /obj/abstract/screen/fullscreen/astral_border)
+		update_fullscreen_alpha("astralborder", 255, 5)
+
+	//astral projections can identify cultists
+	var/datum/faction/bloodcult/cult = find_active_faction_by_type(/datum/faction/bloodcult)
+	if (cult)
+		for(var/datum/role/cultist in cult.members)
+			if(cultist.antag && cultist.antag.current)
+				var/imageloc = cultist.antag.current
+				if(istype(cultist.antag.current.loc,/obj/mecha))
+					imageloc = cultist.antag.current.loc
+				var/hud_icon = cultist.logo_state
+				var/image/I = image('icons/role_HUD_icons.dmi', loc = imageloc, icon_state = hud_icon)
+				I.pixel_x = 20 * PIXEL_MULTIPLIER
+				I.pixel_y = 20 * PIXEL_MULTIPLIER
+				I.plane = ANTAG_HUD_PLANE
+				client.images += I
+
+
+/mob/living/simple_animal/astral_projection/Destroy()
```

Comment:
I don't like the idea of performing any kind of gameplay or player-facing stuff in `Destroy()` itself. Segment the "return to our body" part in some other proc, I'm sure this will save you on copypaste.

Are those `spawn`s really necessary ?